exec/java-exec/src/main/resources/rest/profile/profile.ftl

```diff
@@ -77,16 +84,17 @@ table.sortable thead .sorting_desc { background-image: url("/static/img/black-de
       <p>
 
         <#if model.isOnlyImpersonationEnabled()>
+          <div id="query-editor" class="form-group">${model.getProfile().query}</div>
+          <input class="form-control" id="query" name="query" type="hidden" value="${model.getProfile().query}"/>
           <div class="form-group">
             <label for="userName">User Name</label>
             <input type="text" size="30" name="userName" id="userName" placeholder="User Name" value="${model.getProfile().user}">
           </div>
         </#if>
 
         <form role="form" id="queryForm" action="/query" method="POST">
-          <div class="form-group">
-            <textarea class="form-control" id="query" name="query" style="font-family: Courier;">${model.getProfile().query}</textarea>
-          </div>
+          <div id="query-editor" class="form-group">${model.getProfile().query}</div>
+          <input class="form-control" id="query" name="query" type="hidden" value="${model.getProfile().query}"/>
```

Comment:
It looks like you have the same code on lines 87-88. These code lines are included when only impersonation is enabled. So if I am not mistaken, you'll end up with two query editors when only impersonation is enabled. Please check.

Reply:
Let me take a look. This might have slipped past during a merge conflict.

Reply:
Patched and Verified that with Impersonation enabled, there is only one editor for the profile page.